base/bucket_gocb.go

```diff
@@ -1795,14 +1795,22 @@ func (bucket *CouchbaseBucketGoCB) Incr(k string, amt, def uint64, exp uint32) (
 	// retry handling, so doesn't need redundant retry handling here.
 	if amt == 0 {
 		var result uint64
-		_, err := bucket.Get(k, &result)
+		cas, err := bucket.Get(k, &result)
 		if bucket.IsKeyNotFoundError(err) {
-			// Return Incr value as zero
-			return uint64(0), nil
+			if def != 0 {
+				// try a cas-safe write to initialize the counter to def
+				_, err := bucket.WriteCas(k, 0, 0, cas, def, sgbucket.Raw)
```

Comment:
Since we're doing this inside a KeyNotFound scenario, can we just set cas to zero here?  I think that makes the intention more explicit (only create if it doesn't exist).